From d3b0272f13943dbf61b51dfddbcbab6a8a6433ea Mon Sep 17 00:00:00 2001
From: Andy Holmes <andrew.g.r.holmes@gmail.com>
Date: Mon, 30 Oct 2023 11:36:44 -0700
Subject: [PATCH] online-accounts: fix null pointer de-reference

Ensure goa_account_call_remove() isn't called on a NULL pointer.
---
 panels/online-accounts/cc-online-accounts-panel.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/panels/online-accounts/cc-online-accounts-panel.c b/panels/online-accounts/cc-online-accounts-panel.c
index a889f3c86a..8ac3eaea8e 100644
--- a/panels/online-accounts/cc-online-accounts-panel.c
+++ b/panels/online-accounts/cc-online-accounts-panel.c
@@ -699,10 +699,13 @@ on_client_remove_account_finish_cb (GoaAccount   *account,
 static void
 on_notification_closed_cb (CcOnlineAccountsPanel *self)
 {
-  goa_account_call_remove (goa_object_peek_account (self->removed_object),
-                           cc_panel_get_cancellable (CC_PANEL (self)),
-                           (GAsyncReadyCallback) on_client_remove_account_finish_cb,
-                           g_object_ref (self));
+  if (self->removed_object != NULL)
+    {
+      goa_account_call_remove (goa_object_peek_account (self->removed_object),
+                               cc_panel_get_cancellable (CC_PANEL (self)),
+                               (GAsyncReadyCallback) on_client_remove_account_finish_cb,
+                               g_object_ref (self));
+    }
 
   gtk_revealer_set_reveal_child (self->notification_revealer, FALSE);
 
-- 
GitLab

